server/src/streaming/systems/streams.rs

```diff
@@ -81,6 +82,30 @@ impl System {
         self.streams.values().collect()
     }
 
+    pub fn find_streams(&self, session: &Session) -> Result<Vec<&Stream>, Error> {
+        if !session.is_authenticated() {
+            return Err(Error::Unauthenticated);
+        }
```

Comment:
maybe move this to separate method, i've seen this repeated multiple times in this PR. I suggest something like
```rust
fn ensure_authenticated(&self, session: &Session) -> Result<(), Error> {
    match session.is_authenticated() {
        true => Ok(()),
        false => Err(Error::Unauthenticated),
    }
}
```

Reply:
Good idea, done.